packages/viewer/src/lib/transforms/transformation-delta.ts

```diff
@@ -0,0 +1,77 @@
+import { Matrix4, Quaternion, Vector3 } from '@vertexvis/geometry';
+
+export const ALMOST_ONE = 0.9999;
+/**
+ * There are an infinite number of orthogonal vectors, this function is used to choose one.
+ * @param normal
+ * @returns
+ */
+export function chooseOrthogonalVector(
+  normal: Vector3.Vector3
+): Vector3.Vector3 {
+  const absNorm = Vector3.create(
+    Math.abs(normal.x),
+    Math.abs(normal.y),
+    Math.abs(normal.z)
+  );
+  const x = absNorm.x < absNorm.y && absNorm.x < absNorm.z ? 1.0 : 0.0;
+  const y = absNorm.y <= absNorm.x && absNorm.y < absNorm.z ? 1.0 : 0.0;
+  const z = absNorm.z <= absNorm.x && absNorm.z <= absNorm.y ? 1.0 : 0.0;
+
+  const vector = Vector3.create(x, y, z);
+  return Vector3.normalize(Vector3.cross(normal, vector));
+}
+
+/**
+ * Computes the rotation matrix for two normal vectors.
+ * @param normal1
+ * @param normal2
+ * @returns
+ */
```

Comment:
nit: Same as above here, could we explain a bit further how the rotation matrix is computed? I.e. explain that it's computing the rotation matrix such that both normals are anti-parallel